bl/interfaces/controller.go

```diff
@@ -0,0 +1,47 @@
+package interfaces
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/kimbellG/kerror"
+	ttgrpc "github.com/kimbellG/tournament-bl/interfaces/grpc"
+	"github.com/kimbellG/tournament-bl/models"
+	"github.com/kimbellG/tournament-bl/usecases"
+)
+
+type ServiceController struct {
+	userUsecase usecases.UserUsecase
+}
+
+func (sc *ServiceController) SaveUser(ctx context.Context, user *ttgrpc.User) (*ttgrpc.SaveResponse, error) {
```

Comment:
we don't pass transport layer models to any other layer. I'd suggest you create adapter called internal.User (basically the same model but not protoc generated). Every time you receive grpc model at transport layer (in your case grpc handler), you convert it to adapter (your internal.User model in this case), and you pass than adapter to any layer you want. So basically everything you get from outside (http, grpc models and etc) you convert to internal adapter, and you pass that internal adapter to to any layer. Transport stuff should stay at transport layer.

Reply:
Why? This is presenters layer, it includes implemention of db and transport protocol. This function is part of transport implemention